File: go-cli/internal/handler/update.go
// Package handler contains CLI command handlers.
package handler

import (
	"fmt"

	"convex-notes-cli/internal/service"

	"github.com/AlecAivazis/survey/v2"
	"github.com/fatih/color"
)

// UpdateHandler handles the update note command.
type UpdateHandler struct {
	service *service.NoteService
}

// NewUpdateHandler creates a new UpdateHandler.
func NewUpdateHandler(svc *service.NoteService) *UpdateHandler {
	return &UpdateHandler{service: svc}
}

// Execute runs the handler.
func (h *UpdateHandler) Execute() error {
	blue := color.New(color.FgBlue, color.Bold)
	green := color.New(color.FgGreen)
	yellow := color.New(color.FgYellow)

	blue.Println("\n📝 Update a note")

	notes, err := h.service.List()
	if err != nil {
		return fmt.Errorf("failed to list notes: %w", err)
	}

	if len(notes) == 0 {
		yellow.Println("📭 No notes to update.")
		return nil
	}

	// Build options
	options := make([]string, len(notes))
	for i, note := range notes {
		options[i] = note.Title
	}

	var selectedIdx int
	if err := survey.AskOne(&survey.Select{
		Message: "Select a note to update:",
		Options: options,
	}, &selectedIdx); err != nil {
		return err
	}

	selected := notes[selectedIdx]

	var title, content string
	if err := survey.AskOne(&survey.Input{
		Message: "New title:",
		Default: selected.Title,
	}, &title); err != nil {
		return err
	}

	if err := survey.AskOne(&survey.Input{
		Message: "New content:",
		Default: selected.Content,
	}, &content); err != nil {
		return err
	}

	if err := h.service.Update(selected.ID, title, content); err != nil {
		return fmt.Errorf("failed to update note: %w", err)
	}

	green.Println("\n✅ Note updated successfully!")

	return nil
}

File: python-cli/src/convex_notes/handlers/update.py
"""Update note handler."""

from rich.console import Console
from rich.prompt import Prompt

from convex_notes.service import NoteService

console = Console()


def execute(service: NoteService) -> None:
    """Execute the update note command."""
    console.print("\n[bold blue]📝 Update a note[/bold blue]")

    notes = service.list()

    if not notes:
        console.print("[yellow]📭 No notes to update.[/yellow]")
        return

    # Display notes for selection
    console.print("\nSelect a note to update:")
    for i, note in enumerate(notes, 1):
        console.print(f"  [cyan]{i}[/cyan]. {note.title}")

    try:
        choice = int(Prompt.ask("\n[cyan]Enter number[/cyan]")) - 1
        if choice < 0 or choice >= len(notes):
            console.print("[red]❌ Invalid selection.[/red]")
            return
    except ValueError:
        console.print("[red]❌ Invalid input.[/red]")
        return

    selected = notes[choice]

    title = Prompt.ask("[cyan]New title[/cyan]", default=selected.title)
    content = Prompt.ask("[cyan]New content[/cyan]", default=selected.content)

    try:
        service.update(selected.id, title, content)
        console.print("\n[green]✅ Note updated successfully![/green]")
    except Exception as e:
        console.print(f"[red]❌ Failed to update note:[/red] {e}")
